### DISCUSSION TITLE
Simplest way to upload pre-built sketch from users' Win10 (without installing Arduino)?

### DISCUSSION THREAD

**Discussion by XFer012** [earlephilhower/arduino-pico#1108] [Q&A] [answered]
Hello,
I have a proper working build environment and can build and upload sketches to my Picos just fine (Win10 + Arduino IDE 1.8.9 + arduino-pico 2.6.5).
I need to send a new firmware to a user, which has the Pico board but has not installed the whole Arduino ecosystem.
So basically I build the sketch, send it to him and then he has to upload it (via USB) to his board.

Which is the simplest way for him to upload my sketch? If possible, he would (strongly) prefer not to install Arduino IDE etc., but a command line tool or two would be OK.

Can I, say, copy a couple "self-contained" CLI tools (which ones?) from my system and give them to him? Would this work?
Many thanks!

PS: it would be great to have both case covered: upload to a board running a previous version of my sketch, so already "primed" with Arduino let's say, and to a brand-new board out from the box.

**Comment by earlephilhower** [comment, marked as answer]
Just send the `blah.ino.uf2` file that's generated by the normal build process, and then have the user hold BOOTSEL, plug in the Pico, and copy over the UF2 file.  That requires no tools on the receiving end at all...

**Comment by Andy2No** [comment]
@XFer012 See #288 for a bit more detail - changing your build folder to one you can find easily (covered in that thread) is a useful tip.

**Comment by XFer012** [comment]
Thank you very much, both of you!

**Comment by MYCAMEL222** [comment]
If your BOOTSEL button is not easily assessable (pico in a case or something) you can do the following.

Have the user go to https://googlechromelabs.github.io/serial-terminal/

If you set the baud rate to 1200 and "connect" to the pico, it will enter the BOOT mode without pressing the BOOTSEL button.

From there, the pico will show up as a drive to the user and the .uf2 file can be dragged and dropped by the user..

No need for the user to download any software or mess with Arduino. 

It seems to be working for me...
